PurchasesTests/Caching/DeviceCacheTests.swift

```diff
@@ -37,63 +37,63 @@ class DeviceCacheTests: XCTestCase {
             .to(equal(userID))
     }
 
-    func testClearCachesRemovesCachedPurchaserInfo() {
-        self.deviceCache.clearCaches(forAppUserID: "cesar")
+    func testClearCachesForAppUserIDAndSaveNewUserIDRemovesCachedPurchaserInfo() {
+        self.deviceCache.clearCaches(forAppUserID: "cesar", andSaveNewUserID: "newUser")
         expect(self.mockUserDefaults.removeObjectForKeyCalledValues.contains("com.revenuecat.userdefaults.purchaserInfo.cesar"))
             .to(beTrue())
     }
 
-    func testClearCachesRemovesCachedOfferings() {
+    func testClearCachesForAppUserIDAndSaveNewUserIDRemovesCachedOfferings() {
         let offerings = Purchases.Offerings()
         self.deviceCache.cacheOfferings(offerings)
         expect(self.deviceCache.cachedOfferings).to(equal(offerings))
-        self.deviceCache.clearCaches(forAppUserID: "cesar")
+        self.deviceCache.clearCaches(forAppUserID: "cesar", andSaveNewUserID: "newUser")
         expect(self.deviceCache.cachedOfferings).to(beNil())
     }
 
-    func testClearCachesClearsCachesTimestamp() {
+    func testClearCachesForAppUserIDAndSaveNewUserIDClearsCachesTimestamp() {
         self.deviceCache.setPurchaserInfoCacheTimestampToNow()
-        self.deviceCache.clearCaches(forAppUserID: "cesar")
+        self.deviceCache.clearCaches(forAppUserID: "cesar", andSaveNewUserID: "newUser")
         expect(self.deviceCache.isPurchaserInfoCacheStale()).to(beTrue())
     }
 
-    func testClearCachesRemovesCachedAppUserIDs() {
```

Comment:
This test name wasn't very descriptive, but it was testing the migration to the newer `appUserIDs` we added on Identity 2.0. I think we should keep it since the migration can still be needed.

Reply:
it's not really removed, it's replaced with `testClearCachesForAppUserIDAndSaveNewUserIDUpdatesCachedAppUserID`. 
however, there's a missing line here that should be checking that we remove the legacy appUserID, I'll add it